opendrift/models/basemodel.py

```diff
@@ -5188,65 +5251,68 @@ def add_halo_readers(self):
+
+        if not hasattr(OpenDriftSimulation, '__anim_no__'):
+            OpenDriftSimulation.__anim_no__ = {}
+
+        if caller not in OpenDriftSimulation.__anim_no__:
+            OpenDriftSimulation.__anim_no__[caller] = 0
 
-            if not hasattr(OpenDriftSimulation, '__anim_no__'):
-                OpenDriftSimulation.__anim_no__ = {}
+        os.makedirs(adir, exist_ok=True)
 
-            if caller not in OpenDriftSimulation.__anim_no__:
-                OpenDriftSimulation.__anim_no__[caller] = 0
+        filename = '%s_%d.gif' % (caller,
+                                  OpenDriftSimulation.__anim_no__[caller])
+        OpenDriftSimulation.__anim_no__[caller] += 1
 
-            os.makedirs(adir, exist_ok=True)
+        filename = os.path.join(adir, filename)
 
-            filename = '%s_%d.gif' % (caller,
-                                      OpenDriftSimulation.__anim_no__[caller])
-            OpenDriftSimulation.__anim_no__[caller] += 1
+        return filename
 
-            filename = os.path.join(adir, filename)
+
+    def _save_animation(self, anim, filename, fps, fastwriter=False):
+        from opendrift.export.punkrockwriters import PunkFFMpegWriter, PunkImageMagickWriter#, PunkPillowWriter
+
+        #fastwriter=True
+
+        if filename is None or 'sphinx_gallery' in sys.modules:
+            filename = self._sphinx_gallery_filename(stack_offset=3)
 
         logger.info('Saving animation to ' + filename + '...')
 
+        start_time = datetime.now()
+
         try:
             if filename[-4:] == '.gif':  # GIF
                 logger.info('Making animated gif...')
                 if not fastwriter:
                     anim.save(filename, fps=fps, writer='imagemagick')
                 elif fastwriter:
                     writergif = PunkImageMagickWriter(fps=fps)
+                    # writergif = PunkPillowWriter(fps=fps)
+                    # writergif = animation.ImageMagickFileWriter()
+
                     anim.save(filename, writer=writergif)
             else:  # MP4
                 try:
                     if not fastwriter:
-                        FFwriter = animation.FFMpegWriter(
-                            fps=fps,
-                            codec='libx264',
-                            bitrate=1800,
-                            extra_args=[
-                                '-profile:v',
-                                'baseline',
-                                '-vf',
-                                'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
-                                '-pix_fmt',
-                                'yuv420p',
-                                '-an'
-                            ])
+                        FFwriter=animation.FFMpegWriter
                     elif fastwriter:
-                        FFwriter = PunkFFMpegWriter(
+                        FFwriter = PunkFFMpegWriter
+                    FFwriter = FFwriter(
```

Comment:
This looks like a bug. Which probably also means that this code path is not tested.

Reply:
which is a reminder that we should increase the general test coverage..... 53% is not very impressive https://coveralls.io/github/OpenDrift/opendrift?branch=master

Reply:
I don't think it's a bug, I tested it, it works... but maybe it's not elegant to define FFwriter first as a class and then as an instance? 

                    if not fastwriter:
                        **FFwriter**=animation.FFMpegWriter
                    elif fastwriter:
                        **FFwriter** = PunkFFMpegWriter
                    **FFwriter** = **FFwriter**(
                            fps=fps,
                            codec='libx264',
                            bitrate=1800,
                            extra_args=[
                                '-profile:v',
                                'baseline',
                                '-vf',
                                'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
                                '-pix_fmt',
                                'yuv420p',
                                '-an'
                            ])
                    anim.save(filename, writer=FFwriter)

Reply:
I have tested it it works, did it like this for not writing the same set of arguments two times

                    if not fastwriter:
                        FFwriter=animation.FFMpegWriter
                    elif fastwriter:
                        FFwriter = PunkFFMpegWriter
                    FFwriter = FFwriter(
                            fps=fps,
                            codec='libx264',
                            bitrate=1800,
                            extra_args=[
                                '-profile:v',
                                'baseline',
                                '-vf',
                                'crop=trunc(iw/2)*2:trunc(ih/2)*2',  # cropping 1 pixel if not even
                                '-pix_fmt',
                                'yuv420p',
                                '-an'
                            ])
                    anim.save(filename, writer=FFwriter)

Reply:
ok, I believe this looks fine

Reply:
....though a lazy nitpicker might prefer
```
if fastwriter:
    FFwriter = PunkFFMpegWriter
else:
    FFwriter = animation.FFMpegWriter
```

Reply:
Yeah, it should be changed to `else:`. Which puts me in the `lazy nitpicker` category ;)